ckanext/data_qld/resource_visibility/helpers.py

```diff
@@ -64,27 +64,27 @@ def process_resources(data_dict, user_obj):
     the resource_visibility value and user.
     """
     # Loop each resources,
-    # If resource is `Resource NOT visible/Pending acknowledgement` and
+    # If resource_visibility is `FALSE` or
+    # resource_visibility is `TRUE` and governance_acknowledgement is `NO`
+    # and de_identified_data is `YES` and
     # if user is NOT Members, Editors, Admins or sysadmin, remove the resource.
     is_sysadmin = user_obj is not None and user_obj.sysadmin
-
-    if not is_sysadmin:
-        if not has_user_permission_for_org(data_dict.get('owner_org'), user_obj, 'read'):
-            resources = data_dict.get('resources', [])
-            options = get_select_field_options('resource_visibility')
-            de_identified_data = data_dict.get('de_identified_data', 'NO') == 'YES'
-
-            for resource in list(resources):
-                resource_visibility = resource.get('resource_visibility', '')
-                # Value of options[2] == Resource NOT visible/Pending acknowledgement.
-                if resource_visibility == options[2].get('value') if len(options) >= 3 else False \
-                        or len(resource_visibility) == 0 and de_identified_data:
-                    resources.remove(resource)
-                    data_dict['num_resources'] -= 1
-                else:
-                    # Need to remove the resource visibility field from display
-                    # if current user don't have access to it.
-                    process_resource_visibility(resource)
+    if not is_sysadmin and not has_user_permission_for_org(
+        data_dict.get('owner_org'), user_obj, 'read'
+    ):
+        resources = data_dict.get('resources', [])
+        de_identified_data = data_dict.get('de_identified_data', 'NO')
+        for resource in list(resources):
+            resource_visible = resource.get('resource_visibility', 'FALSE')
+            governance_acknowledgement = resource.get('governance_acknowledgement', 'NO')
+            if (
+                resource_visible == 'TRUE'
```

Comment:
This is a bit confusing and I think it could be more simple with some like the below.
```
resource_visible == 'False'
or (governance_acknowledgement == 'NO'
and de_identified_data == 'YES')
```
Please test the logic